src/AstTransforms.cpp

```diff
@@ -1026,6 +1026,39 @@ bool ReplaceSingletonVariablesTransformer::transform(AstTranslationUnit& transla
     return changed;
 }
 
+bool NameUnnamedVariablesTransformer::transform(AstTranslationUnit& translationUnit) {
+    bool changed = false;
+    static constexpr const char* boundPrefix = "+underscore";
+
+    struct nameVariables : public AstNodeMapper {
+        mutable bool changed;
+        mutable size_t count;
+        nameVariables() : changed(false), count(0) {}
+
+        std::unique_ptr<AstNode> operator()(std::unique_ptr<AstNode> node) const override {
+            if (dynamic_cast<AstUnnamedVariable*>(node.get())) {
```

Comment:
Compare to nullptr, rather than rely on the implicit boolean conversion
```suggestion
            if (nullptr != dynamic_cast<AstUnnamedVariable*>(node.get())) {
```